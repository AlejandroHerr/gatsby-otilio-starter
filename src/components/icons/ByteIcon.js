import React from 'react';
import PropTypes from 'prop-types';

const ByteIcon = ({ children, className, style }) => (
  <svg
    className={className}
    style={{
      width: '.75em',
      height: '.75em',
      fill: 'none',
      stroke: 'currentColor',
      strokeLinecap: 'round',
      strokeLinejoin: 'round',
      strokeWidth: '2',
      ...style,
    }}
    viewBox="0 0 32 32"
    xmlns="http://www.w3.org/2000/svg"
  >
    {children}
  </svg>
);

ByteIcon.defaultProps = {
  className: '',
  style: {},
};

ByteIcon.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  style: PropTypes.object, // eslint-disable-line react/forbid-prop-types
};

export default ByteIcon;
