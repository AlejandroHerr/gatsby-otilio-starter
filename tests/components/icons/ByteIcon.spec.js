import React from 'react';
import { shallow } from 'enzyme';

import ByteIcon from '../../../src/components/icons/ByteIcon';

const defaultProps = {
  icon: 'testIcon',
};

const setup = (props = {}) => {
  const byteIcon = shallow(<ByteIcon {...defaultProps} {...props} />);

  return {
    byteIcon,
  };
};

describe('icons', () => {
  describe('ByteIcon', () => {
    it('should render a default ByteIcon', () => {
      const { byteIcon } = setup();

      expect(byteIcon.hasClass(`byte_icon__${defaultProps.icon}`)).toBeTruthy();
      expect(byteIcon).toMatchSnapshot();
    });
    it('should render a ByteIcon with custom options', () => {
      const props = {
        bold: true,
        className: 'testClassName',
      };
      const { byteIcon } = setup(props);

      expect(byteIcon.hasClass(`byte_icon__${defaultProps.icon}`)).toBeTruthy();
      expect(byteIcon.hasClass('byte_icon__bold')).toBeTruthy();
      expect(byteIcon.hasClass(props.className)).toBeTruthy();
      expect(byteIcon).toMatchSnapshot();
    });
  });
});
