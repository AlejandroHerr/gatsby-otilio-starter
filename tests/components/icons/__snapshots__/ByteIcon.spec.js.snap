// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icons ByteIcon should render a ByteIcon with custom options 1`] = `
<svg
  className="byte_icon byte_icon__testIcon byte_icon__bold testClassName"
  viewBox="0 0 32 32"
  xmlns="http://www.w3.org/2000/svg"
>
  <path />
</svg>
`;

exports[`icons ByteIcon should render a default ByteIcon 1`] = `
<svg
  className="byte_icon byte_icon__testIcon"
  viewBox="0 0 32 32"
  xmlns="http://www.w3.org/2000/svg"
>
  <path />
</svg>
`;
